app/validators/answers/number_answer_validator.py

```diff
@@ -60,6 +61,8 @@ def validate_mandatory_has_no_default(self):
     def validate_value_in_limits(self):
```

Comment:
This can be simplified/fixed to something like:

```py
    def validate_value_in_limits(self):
        min_value = self.answer.get("minimum", {}).get("value", 0)
        max_value = self.answer.get("maximum", {}).get("value", 0)
        answer_ranges = self.questionnaire_schema.numeric_answer_ranges

        if isinstance(min_value, int) and min_value < MIN_NUMBER:
            self.add_error(
                self.MINIMUM_LESS_THAN_LIMIT, value=min_value, limit=MIN_NUMBER
            )
        elif isinstance(min_value, dict):
            source = min_value["source"]
            referred_answer = (
                self.questionnaire_schema.get_numeric_value_for_value_source(
                    source, min_value, answer_ranges
                )
            )
            if referred_answer["min"] < MIN_NUMBER:
                self.add_error(
                    self.MINIMUM_LESS_THAN_LIMIT, value=referred_answer["min"], limit=MIN_NUMBER
                )

        if isinstance(max_value, int) and max_value > MAX_NUMBER:
            self.add_error(
                self.MAXIMUM_GREATER_THAN_LIMIT, value=max_value, limit=MAX_NUMBER
            )
        elif isinstance(min_value, dict):
            source = max_value["source"]
            referred_answer = (
                self.questionnaire_schema.get_numeric_value_for_value_source(
                    source, min_value, answer_ranges
                )
            )
            if referred_answer["max"] > MAX_NUMBER:
                self.add_error(
                    self.MAXIMUM_GREATER_THAN_LIMIT, value=referred_answer["max"], limit=MAX_NUMBER
                )
```

You will need to add tests for value source checks

Reply:
UPDATE TO REVIEWERS: After discussion I am going to add tests in another Pull request later when I have time .